src/sync/stagex.rs

```diff
@@ -58,6 +58,19 @@ impl Manifest {
                     let version = self.version(source)?;
                     version.replace('.', "_")
                 }
+                "version_major" => {
+                    let version = self.version(source)?;
+                    version.split('.').next().unwrap_or("").to_string()
+                }
+                "version_major_minor" => {
+                    let version = self.version(source)?;
+                    let parts: Vec<&str> = version.split('.').collect();
+                    format!("{}.{}", parts[0], parts[1])
```

Comment:
Thanks for working on this! This line could panic the process if the version string is a single number with no dots.

I think in that case it could either use the single number as version string, or return an error (you can extend the enum in `src/errors.rs`, there are already some StageX specific errors defined).

Reply:
No problem, I'll make the adjustment.